editor/src/components/canvas/dom-lookup.ts

```diff
@@ -46,6 +46,27 @@ export function findFirstParentWithValidUID(
   }
 }
 
+export function getValidTargetAtPoint(
+  validTemplatePaths: Array<string>,
+  point: WindowPoint | null,
+): TemplatePath | null {
+  if (point == null) {
+    return null
+  }
+  const elementsUnderPoint = document.elementsFromPoint(point.x, point.y)
+  return (
+    stripNulls(
```

Comment:
This would probably be better formulated as a regular for...of loop that returns the not null case with a default return null after it. Otherwise this is creating two arrays that are immediately thrown away for that first result, after going through two arrays fully.

Reply:
Or just an [`Array.prototype.find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find)

Reply:
It's applying some logic inside this loop however, which is why that can't be used alone.

Reply:
fixed